### DISCUSSION TITLE
how to directly use jwt token embedding dashboard(do not use embedding sdk)

### DISCUSSION THREAD

**Discussion by laoba25** [apache/superset#22108] [Q&A / Help]
hi teams,

Recently, there is a project that wants to integrate the dashboard of superset.
I tried to use the superset embedding sdk, but because it treats all users as the same guest user, I can't use the pre-assigned rls and roles to automatically display the relevant dashboards like logging into the superset directly.

So I tried to embed the dashboard url directly with iframe, but after several days of trying, it still failed.

```js
<script src="https://code.jquery.com/jquery-1.12.4.js" type="text/javascript"></script>

<script type="text/javascript">
    function functionToExecute() {
        var url = "http://localhost:9999/superset/dashboard/6/?standalone=true";
        $.ajax({
            type: 'GET',
            url: url,
            headers: {
                'Authorization': 'Bearer xxx'
            },
            success: function (data) {
                console.log("ok");
                $("#myiframe").css("display", "block")
                $("#myiframe").attr("src", url)
            },
            error: function (error) {
                console.log("err");
                console.log(error);
            },
            statusCode: {
                400: function () {
                    alert("server returned a bad request response.");
                }
            }
        });
    }
</script>
<button id="TestButton" onclick="functionToExecute()">refresh</button>

<iframe id="myiframe"
        style="position: fixed; z-index: 999; border-color:yellow; height: 50%; width: 50%; top: 20%; left:20%;filter: alpha(opacity=60); opacity: 0.6; -moz-opacity: 0.8;display:none"></iframe>

```

after click the refresh button, iframe redirect into login page.
<img width="289" alt="image" src="https://user-images.githubusercontent.com/3195838/201529043-39d4b8f4-1455-45c0-9344-3bbbbd2beece.png">

Ask if there is any good solution.
thank you very much.

**Comment by laoba25** [comment]
Hope someone can provide some info, thanks a lot

**Comment by apole** [comment]
I am looking for similar solution. but could not find a way to set this token for iframe usage.

**Comment by zs-err** [comment]
@dalei2019 did you find solution for this?
